File: src/main/java/org/myfintech/payment/service/PaymentService.java
package org.myfintech.payment.service;

import java.util.List;

import org.myfintech.payment.domain.PaymentDTO;
import org.myfintech.payment.entity.Payment;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface PaymentService {
	Page<PaymentDTO> findAll(Pageable pageable);

	Payment findById(Long id);

	List<PaymentDTO> findPaymentsByContractNumber(String contractNumber);

	List<Payment> savePayments(List<Payment> paymentEntities);

	Payment savePayment(Payment payment, String trackingNumber);

	public void saveTrackedPayments(String trackingNumber, List<Payment> paymentEntities);
	
	public void validate(PaymentDTO dto);

	Payment savePayment(Payment payement);
}

File: src/main/kotlin/org/myfintech/payment/service/PaymentService.kt
package org.myfintech.payment.service

import org.myfintech.payment.domain.PaymentDTO
import org.myfintech.payment.entity.Payment
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface PaymentService {
    fun findAll(pageable: Pageable): Page<PaymentDTO>

    fun findById(id: Long): Payment

    fun findPaymentsByContractNumber(contractNumber: String): List<PaymentDTO>

    fun savePayments(paymentEntities: List<Payment>): List<Payment>

    fun savePayment(payment: Payment, trackingNumber: String): Payment

    fun saveTrackedPayments(trackingNumber: String, paymentEntities: List<Payment>)

    fun validate(dto: PaymentDTO)

    fun savePayment(payement: Payment): Payment
}
